mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageIdMapperTest.java

```diff
@@ -0,0 +1,413 @@
+        notPersistedMailbox.setMailboxId(mapperProvider.generateId());
+        SimpleMailboxMessage message = createMessage(notPersistedMailbox, "Subject: Test \n\nBody\n.\n", BODY_START, new PropertyBuilder());
+        message.setUid(mapperProvider.generateMessageUid());
+
+        expectedException.expect(MailboxNotFoundException.class);
+        sut.save(message);
+    }
+
+    @ContractTest
+    public void saveShouldSaveMessageAnotherIndicesWhenMessageAlreadyInMailbox() throws Exception {
+        message1.setUid(mapperProvider.generateMessageUid());
+        sut.save(message1);
+
+        SimpleMailboxMessage message1InOtherMailbox = SimpleMailboxMessage.copy(benwaWorkMailbox.getMailboxId(), message1);
+        message1InOtherMailbox.setUid(mapperProvider.generateMessageUid());
+        sut.save(message1InOtherMailbox);
+
+        List<MailboxId> mailboxes = sut.findMailboxes(message1.getMessageId());
+        assertThat(mailboxes).containsOnly(benwaInboxMailbox.getMailboxId(), benwaWorkMailbox.getMailboxId());
+    }
+
+    @ContractTest
+    public void saveShouldWorkWhenSavingTwoTimesWithSameMessageIdAndSameMailboxId() throws Exception {
+        message1.setUid(mapperProvider.generateMessageUid());
+        sut.save(message1);
+        SimpleMailboxMessage copiedMessage = SimpleMailboxMessage.copy(message1.getMailboxId(), message1);
+        copiedMessage.setUid(mapperProvider.generateMessageUid());
+        sut.save(copiedMessage);
+
+        List<MailboxId> mailboxes = sut.findMailboxes(message1.getMessageId());
+        assertThat(mailboxes).containsOnly(benwaInboxMailbox.getMailboxId(), benwaInboxMailbox.getMailboxId());
+    }
+
+    @ContractTest
+    public void deleteShouldNotThrowWhenUnknownMessage() {
+        sut.delete(message1.getMessageId());
+    }
+
+    @ContractTest
+    public void deleteShouldDeleteAMessage() throws Exception {
+        message1.setUid(mapperProvider.generateMessageUid());
+        sut.save(message1);
+
+        MessageId messageId = message1.getMessageId();
+        sut.delete(messageId);
+
+        List<MailboxMessage> messages = sut.find(ImmutableList.of(messageId), FetchType.Full);
+        assertThat(messages).isEmpty();
+    }
+
+    @ContractTest
+    public void deleteShouldDeleteMessageIndicesWhenStoredInTwoMailboxes() throws Exception {
+        message1.setUid(mapperProvider.generateMessageUid());
+        sut.save(message1);
+
+        SimpleMailboxMessage message1InOtherMailbox = SimpleMailboxMessage.copy(benwaWorkMailbox.getMailboxId(), message1);
+        message1InOtherMailbox.setUid(mapperProvider.generateMessageUid());
+        sut.save(message1InOtherMailbox);
+
+        MessageId messageId = message1.getMessageId();
+        sut.delete(messageId);
+
+        List<MailboxId> mailboxes = sut.findMailboxes(messageId);
+        assertThat(mailboxes).isEmpty();
+    }
+
```

Comment:
. Remove message: we are not removing them yet, only indices.
. remove all MailboxMessage: i don't understand which kind of test you are waiting for.